Enable parsing of any unknown field under `event_attributes`

When indexing the sample data from [Kata 003 in Chorus](https://github.com/o19s/chorus-OpenSearch-edition/blob/main/katas/003_import_preexisting_event_data.md) the judgments generation fails with a parsing exception (see stacktrace below).

The reason for this apparently is an unknown field in the data (`dwell_time`) which is nested under `event_attributes`. According to the specification any additional property is allowed, so handling any potentially existing field should be implemented.

Steps to reproduce:
1. OpenSearch with UBI plugin up & running
2. Index sample data as described in [Kata 003](https://github.com/o19s/chorus-OpenSearch-edition/blob/main/katas/003_import_preexisting_event_data.md)
3. Run judgments generation script  
4. Observe the error

```
Exception in thread "main" java.lang.RuntimeException: Jackson exception
	at org.opensearch.client.transport.httpclient5.ApacheHttpClient5Transport.extractAndWrapCause(ApacheHttpClient5Transport.java:1163)
	at org.opensearch.client.transport.httpclient5.ApacheHttpClient5Transport.performRequest(ApacheHttpClient5Transport.java:158)
	at org.opensearch.client.opensearch.OpenSearchClient.search(OpenSearchClient.java:1386)
	at org.opensearch.eval.engine.OpenSearchEngine.getClickthroughRate(OpenSearchEngine.java:572)
	at org.opensearch.eval.judgments.clickmodel.coec.CoecClickModel.calculateJudgments(CoecClickModel.java:64)
	at org.opensearch.eval.App.main(App.java:98)
Caused by: jakarta.json.JsonException: Jackson exception
	at org.opensearch.client.json.jackson.JacksonUtils.convertException(JacksonUtils.java:51)
	at org.opensearch.client.json.jackson.JacksonJsonpMapper$JacksonValueParser.deserialize(JacksonJsonpMapper.java:140)
	at org.opensearch.client.json.JsonpDeserializer.deserialize(JsonpDeserializer.java:87)
	at org.opensearch.client.json.JsonpMapperBase.deserialize(JsonpMapperBase.java:101)
	at org.opensearch.client.json.JsonpDeserializer$1.deserialize(JsonpDeserializer.java:110)
	at org.opensearch.client.json.NamedDeserializer.deserialize(NamedDeserializer.java:77)
	at org.opensearch.client.json.ObjectDeserializer$FieldObjectDeserializer.deserialize(ObjectDeserializer.java:81)
	at org.opensearch.client.json.ObjectDeserializer.deserialize(ObjectDeserializer.java:185)
	at org.opensearch.client.json.ObjectDeserializer.deserialize(ObjectDeserializer.java:146)
	at org.opensearch.client.json.ObjectBuilderDeserializer.deserialize(ObjectBuilderDeserializer.java:97)
	at org.opensearch.client.json.JsonpDeserializerBase$ArrayDeserializer.deserialize(JsonpDeserializerBase.java:343)
	at org.opensearch.client.json.JsonpDeserializerBase$ArrayDeserializer.deserialize(JsonpDeserializerBase.java:308)
	at org.opensearch.client.json.JsonpDeserializer.deserialize(JsonpDeserializer.java:87)
	at org.opensearch.client.json.ObjectDeserializer$FieldObjectDeserializer.deserialize(ObjectDeserializer.java:81)
	at org.opensearch.client.json.ObjectDeserializer.deserialize(ObjectDeserializer.java:185)
	at org.opensearch.client.json.ObjectDeserializer.deserialize(ObjectDeserializer.java:146)
	at org.opensearch.client.json.JsonpDeserializer.deserialize(JsonpDeserializer.java:87)
	at org.opensearch.client.json.ObjectBuilderDeserializer.deserialize(ObjectBuilderDeserializer.java:91)
	at org.opensearch.client.json.ObjectDeserializer$FieldObjectDeserializer.deserialize(ObjectDeserializer.java:81)
	at org.opensearch.client.json.ObjectDeserializer.deserialize(ObjectDeserializer.java:185)
	at org.opensearch.client.json.ObjectDeserializer.deserialize(ObjectDeserializer.java:146)
	at org.opensearch.client.json.JsonpDeserializer.deserialize(JsonpDeserializer.java:87)
	at org.opensearch.client.json.ObjectBuilderDeserializer.deserialize(ObjectBuilderDeserializer.java:91)
	at org.opensearch.client.json.DelegatingDeserializer$SameType.deserialize(DelegatingDeserializer.java:55)
	at org.opensearch.client.transport.endpoints.EndpointWithResponseMapperAttr$1.deserialize(EndpointWithResponseMapperAttr.java:68)
	at org.opensearch.client.transport.httpclient5.ApacheHttpClient5Transport.decodeResponse(ApacheHttpClient5Transport.java:676)
	at org.opensearch.client.transport.httpclient5.ApacheHttpClient5Transport.prepareResponse(ApacheHttpClient5Transport.java:561)
	at org.opensearch.client.transport.httpclient5.ApacheHttpClient5Transport.lambda$performRequestAsync$0(ApacheHttpClient5Transport.java:191)
	at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:646)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
	at java.base/java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2179)
	at org.opensearch.client.transport.httpclient5.ApacheHttpClient5Transport$1.completed(ApacheHttpClient5Transport.java:236)
	at org.opensearch.client.transport.httpclient5.ApacheHttpClient5Transport$1.completed(ApacheHttpClient5Transport.java:225)
	at org.apache.hc.core5.concurrent.BasicFuture.completed(BasicFuture.java:148)
	at org.apache.hc.core5.concurrent.ComplexFuture.completed(ComplexFuture.java:72)
	at org.apache.hc.client5.http.impl.async.InternalAbstractHttpAsyncClient$2$1.completed(InternalAbstractHttpAsyncClient.java:310)
	at org.apache.hc.core5.http.nio.support.AbstractAsyncResponseConsumer$1.completed(AbstractAsyncResponseConsumer.java:101)
	at org.apache.hc.core5.http.nio.entity.AbstractBinAsyncEntityConsumer.completed(AbstractBinAsyncEntityConsumer.java:87)
	at org.apache.hc.core5.http.nio.entity.AbstractBinDataConsumer.streamEnd(AbstractBinDataConsumer.java:83)
	at org.apache.hc.core5.http.nio.support.AbstractAsyncResponseConsumer.streamEnd(AbstractAsyncResponseConsumer.java:142)
	at org.apache.hc.client5.http.impl.async.HttpAsyncMainClientExec$1.streamEnd(HttpAsyncMainClientExec.java:284)
	at org.apache.hc.core5.http.impl.nio.ClientHttp1StreamHandler.dataEnd(ClientHttp1StreamHandler.java:276)
	at org.apache.hc.core5.http.impl.nio.ClientHttp1StreamDuplexer.dataEnd(ClientHttp1StreamDuplexer.java:366)
	at org.apache.hc.core5.http.impl.nio.AbstractHttp1StreamDuplexer.onInput(AbstractHttp1StreamDuplexer.java:338)
	at org.apache.hc.core5.http.impl.nio.AbstractHttp1IOEventHandler.inputReady(AbstractHttp1IOEventHandler.java:64)
	at org.apache.hc.core5.http.impl.nio.ClientHttp1IOEventHandler.inputReady(ClientHttp1IOEventHandler.java:41)
	at org.apache.hc.core5.reactor.InternalDataChannel.onIOEvent(InternalDataChannel.java:143)
	at org.apache.hc.core5.reactor.InternalChannel.handleIOEvent(InternalChannel.java:51)
	at org.apache.hc.core5.reactor.SingleCoreIOReactor.processEvents(SingleCoreIOReactor.java:176)
	at org.apache.hc.core5.reactor.SingleCoreIOReactor.doExecute(SingleCoreIOReactor.java:125)
	at org.apache.hc.core5.reactor.AbstractSingleCoreIOReactor.execute(AbstractSingleCoreIOReactor.java:92)
	at org.apache.hc.core5.reactor.IOReactorWorker.run(IOReactorWorker.java:44)
	at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException: Unrecognized field "dwell_time" (class org.opensearch.eval.model.ubi.event.EventAttributes), not marked as ignorable (3 known properties: "position", "session_id", "object"])
 at [Source: REDACTED (`StreamReadFeature.INCLUDE_SOURCE_IN_LOCATION` disabled); line: 1, column: 451503] (through reference chain: org.opensearch.eval.model.ubi.event.UbiEvent["event_attributes"]->org.opensearch.eval.model.ubi.event.EventAttributes["dwell_time"])
	at com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException.from(UnrecognizedPropertyException.java:61)
	at com.fasterxml.jackson.databind.DeserializationContext.handleUnknownProperty(DeserializationContext.java:1153)
	at com.fasterxml.jackson.databind.deser.std.StdDeserializer.handleUnknownProperty(StdDeserializer.java:2241)
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.handleUnknownProperty(BeanDeserializerBase.java:1821)
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.handleUnknownVanilla(BeanDeserializerBase.java:1799)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:316)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:177)
	at com.fasterxml.jackson.databind.deser.impl.MethodProperty.deserializeAndSet(MethodProperty.java:129)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:310)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:177)
	at com.fasterxml.jackson.databind.deser.DefaultDeserializationContext.readRootValue(DefaultDeserializationContext.java:342)
	at com.fasterxml.jackson.databind.ObjectMapper._readValue(ObjectMapper.java:4893)
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3036)
	at org.opensearch.client.json.jackson.JacksonJsonpMapper$JacksonValueParser.deserialize(JacksonJsonpMapper.java:138)
	... 51 more
```

--- a/src/main/java/org/opensearch/eval/model/ubi/event/EventAttributes.java
+++ b/src/main/java/org/opensearch/eval/model/ubi/event/EventAttributes.java
@@ -8,8 +8,12 @@
  */
 package org.opensearch.eval.model.ubi.event;
 
+import com.fasterxml.jackson.annotation.JsonAnySetter;
 import com.fasterxml.jackson.annotation.JsonProperty;
 import com.google.gson.annotations.SerializedName;
+
+import java.util.HashMap;
+import java.util.Map;
 
 /**
  * Attributes on an UBI event.
@@ -25,6 +29,8 @@ public class EventAttributes {
 
     @SerializedName("position")
     private Position position;
+
+    private final Map<String, Object> additionalProperties = new HashMap<>();
 
     /**
      * Creates a new object.
@@ -79,6 +85,24 @@ public class EventAttributes {
      */
     public void setPosition(Position position) {
         this.position = position;
+    }
+
+    /**
+     * Adds an unrecognized property to the additional properties map.
+     * @param key   The property name.
+     * @param value The property value.
+     */
+    @JsonAnySetter
+    public void addAdditionalProperty(String key, Object value) {
+        this.additionalProperties.put(key, value);
+    }
+
+    /**
+     * Gets the additional properties.
+     * @return The additional properties map.
+     */
+    public Map<String, Object> getAdditionalProperties() {
+        return additionalProperties;
     }
 
 }

--- a/src/main/java/org/opensearch/eval/model/ubi/event/Position.java
+++ b/src/main/java/org/opensearch/eval/model/ubi/event/Position.java
@@ -8,8 +8,12 @@
  */
 package org.opensearch.eval.model.ubi.event;
 
+import com.fasterxml.jackson.annotation.JsonAnySetter;
 import com.fasterxml.jackson.annotation.JsonProperty;
 import com.google.gson.annotations.SerializedName;
+
+import java.util.HashMap;
+import java.util.Map;
 
 /**
  * A position represents the location of a search result in an event.
@@ -18,6 +22,8 @@ public class Position {
 
     @SerializedName("ordinal")
     private int ordinal;
+
+    private final Map<String, Object> additionalProperties = new HashMap<>();
 
     @Override
     public String toString() {
@@ -40,4 +46,22 @@ public class Position {
         this.ordinal = ordinal;
     }
 
+    /**
+     * Adds an unrecognized property to the additional properties map.
+     * @param key   The property name.
+     * @param value The property value.
+     */
+    @JsonAnySetter
+    public void addAdditionalProperty(String key, Object value) {
+        this.additionalProperties.put(key, value);
+    }
+
+    /**
+     * Gets the additional properties.
+     * @return The additional properties map.
+     */
+    public Map<String, Object> getAdditionalProperties() {
+        return additionalProperties;
+    }
+    
 }

--- a/src/main/java/org/opensearch/eval/model/ubi/event/Position.java
+++ b/src/main/java/org/opensearch/eval/model/ubi/event/Position.java
@@ -9,7 +9,6 @@
 package org.opensearch.eval.model.ubi.event;
 
 import com.fasterxml.jackson.annotation.JsonAnySetter;
-import com.fasterxml.jackson.annotation.JsonProperty;
 import com.google.gson.annotations.SerializedName;
 
 import java.util.HashMap;
@@ -63,5 +62,5 @@ public class Position {
     public Map<String, Object> getAdditionalProperties() {
         return additionalProperties;
     }
-    
+
 }

